refactor(store)!: more reworks (#1860)
tauri-apps/plugins-workspace: .changes/store-plugin-rework.md

---
"store": minor:breaking
---

> **Breaking changes**:

Reviewer: ```suggestion
### Breaking changes:
```

> **Breaking change**: Renamed `StoreCollection` to `StoreState`

Reviewer: We can no longer make breaking changes, we need to revert this or keep it as an alias with a deprecation warning.

Author: Isn't the stability of plugins are still undetermined?

Changes:

> - Disallow calling `create_store` when there're still active stores with the same path alive

Reviewer: Can't we just return the active one then?

Author: That is in `createOrExistingStore` now

Reviewer: yeah, I was talking in terms of breaking changes cause this sounds like another breaking change or a bug fix, debatable tbh.

> Other Changes:

Reviewer: ```suggestion
### Enhancements:
```

- Save and cancel pending auto save on drop
- Use absolute path as store's key, fix #984
- Share store to resource table by default
- Enable auto save with 100ms debounce time by default
- Use pretty json by default, close #1690

> New features:

Reviewer: ```suggestion
### New features:
```

> - Add `getStore`/`get_store` share stores across js and rust side

Reviewer: ```suggestion
- Add `get_store` to get shared stores across js and rust side
```

> - Add default (de)serialize functions settings

Reviewer: should mention the new APIs instead

> - Allow js to use pre-stored (de)serialize functions

Reviewer: this is covered by the JS change file, no need to mention here I guess

Reviewer: or rather mention the relevant public Rust APIs that are used to register these functions

> - Add back lazy store (implemented in js)

Reviewer: this doesn't belong here
